core/trino-main/src/test/java/io/trino/sql/query/QueryAssertions.java

```diff
@@ -475,6 +478,27 @@ public final QueryAssert isNotFullyPushedDown(PlanMatchPattern retainedSubplan)
         {
             PlanMatchPattern expectedPlan = PlanMatchPattern.anyTree(retainedSubplan);
 
+            return hasPlan(expectedPlan, plan -> {
+                if (PlanNodeSearcher.searchFrom(plan.getRoot())
+                        .whereIsInstanceOfAny(TableScanNode.class)
+                        .findFirst().isEmpty()) {
+                    throw new IllegalArgumentException("Incorrect use of isNotFullyPushedDown: the actual plan matched the expected despite not having a TableScanNode left " +
+                            "in the plan. Use hasPlan() instead");
+                }
+            });
+        }
+
+        /**
+         * Verifies query has the expected plan and verifies the results are the same as when the pushdown is fully disabled.
```

Comment:
Verifies query has the expected plan and results are the same, when pushdown is fully disabled.